### DISCUSSION TITLE
Configuration for soft_wrap that excludes indentation

### DISCUSSION THREAD

**Discussion by xpe** [zed-industries/zed#17601] [Feature Requests]
## Background

Some language style guides, including Rust, recommend wrapping at `N` characters **excluding indentation**.

For example, from https://doc.rust-lang.org/beta/style-guide/index.html (with bold text added by me):

> Source lines which are entirely a comment should be limited to 80 characters in length (including comment sigils, but **excluding indentation**) or the maximum width of the line (including comment sigils and indentation), whichever is smaller ...

## Current `soft_wrap` config options

At the time of this comment, here are the Zed config options for `soft_wrap`:

> [Soft Wrap](https://zed.dev/docs/configuring-zed#soft-wrap)
>
> * Description: Whether or not to automatically wrap lines of text to fit editor / preferred width.
> * Setting: `soft_wrap`
> * Default: `prefer_line`
>
> Options
>
> 1. `none` to stop the soft-wrapping
> 2. `prefer_line` to avoid wrapping generally, unless the line is too long
> 3. `editor_width` to wrap lines that overflow the editor width
> 4. `preferred_line_length` to wrap lines that overflow preferred_line_length config value

## Observations

Assuming there was interest, what would be some good ways to extend `soft_wrap` to support Rust's style guide? I don't have a particular recommendation in mind yet. But I can make two observations:

1. It seems to me that `soft_wrap` would need to be aware of language syntax -- in particular, is a line a comment line?

2. `soft_wrap` would also need to have a setting to tell it to ignore indentation.

3. This is starting to feel a little complicated...

4. I wonder ... how do other other editors / code-formatters handle this?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
